src/librustc_yk_sections/emit_tir.rs

```diff
@@ -49,35 +46,37 @@ struct ConvCx<'a, 'tcx, 'gcx> {
     /// The compiler's god struct. Needed for queries etc.
     tcx: &'a TyCtxt<'a, 'tcx, 'gcx>,
     /// Monotonically increasing number used to give TIR variables a unique ID.
-    next_tir_var: Cell<TirLocal>,
+    next_tir_var: Cell<ykpack::LocalIndex>,
     /// A mapping from MIR variables to TIR variables.
-    var_map: RefCell<IndexVec<Local, Option<TirLocal>>>,
```

Comment:
I'm curious why this is a `RefCell`. From what I can see, it's only changed as part of `tir_val()`. Did you have difficulties fighting the borrow checker accepting a `&mut self` reference to `ConvCx` here? Or was it conscious design decision?

If the latter, I think I would prefer passing an explicit mutable reference to `ConvCx` over interior mutability. It's easier to tell that calling this function _will_ cause an internal state change.

Reply:
I commented this on the minus diff by mistake, but I meant it on the new `var_map` field def too ;)

Reply:
This was a good spot. The `RefCell` is left over from before. I don't think it's needed.

Commit coming...